### PR TITLE
Use $BENTOML_PORT as Dockerfile default port

### PR DESCRIPTION
**Pull request by jjmachan**
## Description

This fixes the bug mentioned in #1588. Also checks if $PORT is present
and overides $BENTOML_PORT to $PORT if found.

## Motivation and Context

Solves #1588 
## How Has This Been Tested?

I've tested it locally and on Heroku service. 

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [x] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [x] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by jjmachan** [comment]
Also, I was wondering if I should mention this in the cli reference as well. Right now in effect, we have to environment variables that can be set to specify the port to which our docker service will listen to.

File: bentoml/saved_bundle/templates.py [hunk 1]
```diff
@@ -109,9 +109,9 @@
 # copy over model files
 COPY --chown=bentoml:bentoml . ./
 
-# the env var $PORT is required by heroku container runtime
-ENV PORT 5000
-EXPOSE $PORT
+# Default port for service
+ENV BENTOML_PORT 5000
+EXPOSE $BENTOML_PORT
```
Comment (parano) [inline bentoml/saved_bundle/templates.py:114]: I think we also need `EXPOSE $PORT` here in order for it to work with Heroku right?  are you sure this change works when you also set the `port` to a different value on heroku?

File: docs/source/deployment/heroku.rst [hunk 2]
```diff
@@ -105,6 +105,10 @@ BentoService and available for sending test request:
 Build and deploy to Heroku
 ==========================
 
+.. note::
+  Apps deployed on Heroku must listen to a specific port number which is
+  specified by the $PORT environment variable in the dyno. Bentoml will
```
Comment (parano) [inline docs/source/deployment/heroku.rst:110]: is it a typo?  I guess you mean `...in the demo`?

File: bentoml/saved_bundle/templates.py (hunk 1 above)
Comment (parano) [reply bentoml/saved_bundle/templates.py:114]: never mind, I think I misunderstood the `EXPOSE` command. Looking at related documentation here https://docs.docker.com/engine/reference/builder/#expose, I think it's ok if we just remove `ENV BENTOML_PORT 5000` here and change this line to `EXPOSE 5000`, that way we only define the default port once in the BentoML config, what do you think?

File: docs/source/deployment/heroku.rst (hunk 2 above)
Comment (jjmachan) [reply docs/source/deployment/heroku.rst:110]: no dyno is what heroku calls their containers, so I thought I'd use the same term here.

File: bentoml/saved_bundle/templates.py (hunk 1 above)
Comment (jjmachan) [reply bentoml/saved_bundle/templates.py:114]: I'm not sure. So one doubt is, suppose we use this with something like a docker-compose and override the BENTOML_PORT with  the environment variable to another port, should we also use EXPOSE on that port number or is EXPOSE suppose to be always a fixed port number?

Comment (parano) [reply bentoml/saved_bundle/templates.py:114]: @jjmachan per the linked documentation, EXPOSE doesn't actually do anything: `The EXPOSE instruction does not actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container, about which ports are intended to be published. To actually publish the port when running the container, use the -p flag on docker run to publish and map one or more ports`

If the person who runs the container overrides it via  BENTOML_PORT env var, this person should also be aware to use `-p` to publish that port.  In other cases, `EXPOSE 5000` tells people that this is the default port that will be used.

Comment (jjmachan) [reply bentoml/saved_bundle/templates.py]: Thanks for explaining it @parano! I've made the changes

**Comment by parano** [comment]
thank you @jjmachan!

**Pull request merged**